codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules/EndpointRulesSpecUtils.java

```diff
@@ -207,4 +211,110 @@ public boolean isDeclaredParam(String paramName) {
         Map<String, ParameterModel> parameters = intermediateModel.getEndpointRuleSetModel().getParameters();
         return parameters.containsKey(paramName);
     }
+
+    public FieldSpec parameterClassField(String name, ParameterModel model) {
```

Comment:
I hate to ask this, since there's a lot, but can you javadoc the parameter* methods? It's hard for me to tell what each one is for without looking where it is used.